projects/plugins/jetpack/tests/e2e/specs/connection.test.js

```diff
@@ -15,33 +20,49 @@ process.env.SKIP_CONNECT = true;
  * @group connection
  */
 describe( 'Connection', () => {
-	beforeAll( async () => {
-		await execMultipleWpCommands(
-			'wp option delete e2e_jetpack_plan_data',
-			'wp option delete jetpack_active_plan',
-			'wp option delete jetpack_private_options',
-			'wp option delete jetpack_sync_error_idc'
-		);
-		await page.reload();
-		await page.reload();
+	afterEach( async () => {
+		await execShellCommand( 'yarn tunnel-reset' );
+		global.siteUrl = resolveSiteUrl();
+		await resetWordpressInstall();
+		await loginToWpComIfNeeded( 'defaultUser', true );
+		await loginToWpSite( true );
 	} );
 
-	beforeEach( async () => {
-		await DashboardPage.visit( page );
+	it( 'In-place', async () => {
+		await step( 'Can start in-place connection', async () => {
+			await DashboardPage.visit( page );
+			await ( await Sidebar.init( page ) ).selectJetpack();
+			await doInPlaceConnection();
+		} );
+
+		await step( 'Can assert that site is connected', async () => {
+			const jetpackPage = await JetpackPage.init( page );
+			expect( await jetpackPage.isConnected() ).toBeTruthy();
+		} );
 	} );
 
-	afterAll( async () => {
-		await execWpCommand(
-			`'wp option update jetpack_private_options --format=json < ${ path.resolve(
-				config.get( 'temp.jetpackPrivateOptions' )
-			) }'`
-		);
+	it( 'User-less', async () => {
+		await step( 'Can log out from WPCOM', async () => {
```

Comment:
Maybe deleting the cookies will be a less flaky way to ensure user is logged out.

Reply:
Unfortunately, I was not been able to find an easy way to delete a single cookie. I also tried adding an invalid cookie, but apparently, it does not work as expected.

Do you have ideas/suggestions on how to do that?

Reply:
Also, The cookie should be added back after the test, which should be done in `*Each` hook. The logic there might get tricky

Reply:
I see now that you're using `afterEach`. I wouldn't login back after tests. You don't know if other tests are running after and may execute another login for nothing. Just in case other suites are running after, all suites are running the `maybePreConnect` in `beforeAll` that will perform login if user is logged out anyway.

How about putting the login methods in beforeEach?
And then the userless test deletes all cookies and refreshes the page. 
Also, moving the userless to the end of the spec might reduce the need of log in actions.

Reply:
Moved login steps into beforeEach, and switched to cookie approach as you suggested.